packages/effect/dtslint/Effect.ts

```diff
@@ -14,6 +14,28 @@ declare const numberRecord: Record<string, Effect.Effect<"dep-4", "err-4", numbe
 // forEach
 // -------------------------------------------------------------------------------------
 
+// $ExpectType Effect<"dep-1" | "dep-2", "err-1" | "err-2", string>
+Effect.tap(string, (
+  // $ExpectType string
+  _a
+) => number)
+
+// $ExpectType Effect<"dep-1" | "dep-2", "err-1" | "err-2", string>
+Effect.tap(string, number)
+
+// $ExpectType Effect<"dep-1", "err-1" | UnknownException, string>
+Effect.tap(string, async (
+  // $ExpectType string
+  _a
+) => console.log(""))
+
+// @ts-expect-error
+Effect.tap(string, () => 1)
```

Comment:
This should be valid code, as we want to allow `Effect.tap((_) => console.log(_))` / `Effect.tap((_) => map.delete(_))` / `Effect.tap((_) => postMessage(_))` / whatever (perform sync side effects without having to wrap with Effect.sync)

This was the primary reason we changed the .tap signature.

Reply:
Hmm, now I see the use case for it.
Unfortunately, I won't be able to restrict mistakes like this:
```typescript
const log1 = () => Console.log(1)
Effect.tap(Effect.succeed(1), () => log1) // forgot to call the function
```
But ok, I can live with that.